gax-java/gax-httpjson/src/main/java/com/google/api/gax/httpjson/ProtoMessageRequestFormatter.java

```diff
@@ -95,10 +95,28 @@ public String getRequestBody(RequestT apiMessage) {
     return requestBodyExtractor.extract(apiMessage);
   }
 
-  /* {@inheritDoc} */
+  /**
+   * Returns the relative URL path created from the path parameters from the given message. Attempts
+   * to match the with the default PathTemplate. If there is not match, it attempts to match with
+   * the templates in the additionalPathTemplates.
+   *
+   * @param apiMessage Request object to extract fields from
+   * @return Path of a matching URL
+   */
   @Override
   public String getPath(RequestT apiMessage) {
-    return pathTemplate.instantiate(pathVarsExtractor.extract(apiMessage));
+    String path = pathTemplate.instantiate(pathVarsExtractor.extract(apiMessage));
+    if (pathTemplate.matches(path)) {
+      return path;
+    }
+    for (PathTemplate additionalPathTemplate : additionalPathTemplates) {
+      String additionalPath =
+          additionalPathTemplate.instantiate(pathVarsExtractor.extract(apiMessage));
+      if (additionalPathTemplate.matches(additionalPath)) {
+        return additionalPath;
+      }
+    }
+    throw new IllegalArgumentException("No matching paths for Request: " + apiMessage);
```

Comment:
Was it silently failing before? If it was, throwing an exception now could be considered a breaking change.

Reply:
Yeah, it would making a request to the invalid URL and would result in a 404. I can update this return the default path's raw string instead of the exception.